### PR TITLE
ci: migrate to GitHub actions

### PR DESCRIPTION
**Pull request by KaiPrince**
[US155092](https://rally1.rallydev.com/#/?detail=/userstory/708743294119&fdp=true): Update siren library for radio buttons

**Context**
Our Appveyor CI is broken and must be retired. This PR migrates our CI to Github Actions.

**Quality**
- Repo settings PR: https://github.com/Brightspace/repo-settings/pull/1191
- Failed Run: https://ci.appveyor.com/project/Brightspace/d2l-hypermedia-siren/builds/47552998

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  build:
+    name: Build
+    runs-on: [self-hosted, AWS, Linux]
+    timeout-minutes: 10
+    steps:
+      - uses: Brightspace/third-party-actions@actions/checkout
+      - name: Setup .NET Core
+        uses:  Brightspace/third-party-actions@actions/setup-dotnet
+        with:
+          dotnet-version: 7.0.x
```
Comment (KaiPrince) [inline .github/workflows/main.yml:14]: Most of our other libraries are on v6, but my local machine has v7 installed and everything compiles. Any reason not to upgrade?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Main
```
Comment (KaiPrince) [inline .github/workflows/main.yml:1]: Our other libraries use build, but I've been noticing a pattern of using `main` for the only or primary workflow.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Main
+on: [push]
+
+jobs:
+  build:
+    name: Build
+    runs-on: [self-hosted, AWS, Linux]
```
Comment (KaiPrince) [inline .github/workflows/main.yml:7]: Should this be `ubuntu-latest`?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: dotnet test
+        run: dotnet test D2L.Hypermedia.Siren.Tests --configuration Release --no-build
+
+      - name: dotnet pack
+        run: dotnet pack --configuration Release --no-build
+
+      - name: Publish to CodeArtifact
+        if: github.ref == 'refs/heads/master'
+        run: dotnet nuget push "**/*.symbols.nupkg" --source d2l/private --api-key aws
```
Comment (KaiPrince) [inline .github/workflows/main.yml:38]: We could use [this](https://github.com/Brightspace/D2L.CodeStyle/blob/main/.github/workflows/build.yml#L28) instead, but I'm not sure there's a difference.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      - uses: Brightspace/third-party-actions@NuGet/setup-nuget 
+ 
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+
+      - name: Install dependencies
+        run: nuget restore D2L.Hypermedia.Siren
+
+      - name: Build
+        run: msbuild -p:Configuration=Release D2L.Hypermedia.Siren
```
Comment (KaiPrince) [inline .github/workflows/main.yml:29]: I would rather not use `msbuild`, but nothing else works.  All combinations of `dotnet restore`, `dotnet build` and `dotnet msbuild` fail with this error.

image

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+
+      - uses: Brightspace/third-party-actions@actions/checkout
+      - uses: Brightspace/third-party-actions@actions/setup-dotnet
+        with:
+          dotnet-version: 7.0.x
+      - uses: Brightspace/third-party-actions@NuGet/setup-nuget 
+ 
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
```
Comment (KaiPrince) [inline .github/workflows/main.yml:23]: I'm hoping to find an alternative to `msbuild`, but if not I will add it to our third-party-actions repo before merging this.

File: .github/workflows/main.yml (hunk 5 above)
Comment (AntonBazhal) [reply .github/workflows/main.yml:29]: Looks like the `.csproj` was created in VS and not with `dotnet new`, so the format is slightly different and `dotnet restore` can't find packages. Also, `dotnet restore` does not support `packages.config` that is used in this repo.

I could not make the build work with the current master, but when I switched to the commit prior to [this](https://github.com/Brightspace/D2L.Hypermedia.Siren/pull/36/files) change, I was able to build successfully with 
```
nuget restore
dotnet build -c Release
```

Looks like dbot updated `dotnet restore` but the [.csproj](https://github.com/Brightspace/D2L.Hypermedia.Siren/blob/c61262a22c5bd8f7c04e452f55fbec7019a6c149/D2L.Hypermedia.Siren/D2L.Hypermedia.Siren.csproj#L35) is still pointing to the old version.

**Review by AntonBazhal — APPROVED** [review]

**Pull request merged**